src/controllers/member.js

```diff
@@ -1,155 +1,39 @@
-const express = require('express');
-
-const fs = require('fs');
-
-const members = require('../data/member.json');
-
-const apiMembers = express.Router();
-
-// const router = express.Router();
-
-// Get by ID
-apiMembers.get('/getById/:id', (req, res) => {
-  const memberId = req.params.id;
-  const foundMember = members.find((member) => member.id.toString() === memberId);
-  if (foundMember) {
-    res.send(foundMember);
-  } else {
-    res.send('There is not any member with that ID!');
-  }
-});
-
-apiMembers.post('/post', (req, res) => {
-  const newMember = req.body;
-  // valid empty fields
-  const requiredFields = ['first_name', 'last_name', 'email', 'password', 'birthdate', 'city', 'address', 'phone',
-    'memberships'];
-  const missingFields = requiredFields.filter((field) => !newMember[field]);
-
-  if (missingFields.length > 0) {
-    res.send(`Missing fields: ${missingFields.join(', ')}`);
-    return;
-  }
-
-  members.push(newMember);
-  fs.writeFile('src/data/member.json', JSON.stringify(members, null, 2), (error) => {
-    if (error) {
-      res.send('There was a mistake! Member cannot be created.');
-    } else {
-      res.send('Member created.');
-    }
-  });
-});
-// Delete by ID
-apiMembers.delete('/deleteById/:id', (req, res) => {
-  const memberId = req.params.id;
-  const memberIndex = members.findIndex((member) => member.id.toString() === memberId);
-
-  if (memberIndex !== -1) {
-    members.splice(memberIndex, 1);
-    fs.writeFile('src/data/member.json', JSON.stringify(members, null, 2), (err) => {
-      if (err) {
-        res.send('There was a mistake! Member cannot be deleted.');
-      } else {
-        res.send('Member deleted.');
+const Member = require('../models/Member');
+
+const updateMember = (req, res) => {
+  const { id } = req.params;
+  const {
+    firstName, lastName, dni, birthday, phone, email, city, postalCode, isActive, membership,
+  } = req.body;
+
+  Member.findByIdAndUpdate(
+    id,
+    {
+      firstName,
+      lastName,
+      dni,
+      birthday,
+      phone,
+      email,
+      city,
+      postalCode,
+      isActive,
+      membership,
+    },
+    { new: true },
+  )
+    .then((result) => {
+      if (!result) {
+        return res.status(404).json({
```

Comment:
this should be a 404

Reply:
is a 404, or should it be a 400?

Reply:
it's a 404 error because the member doesn't exists